hack/generate-authors.sh

```diff
@@ -1,15 +1,19 @@
 #!/usr/bin/env bash
+
 set -e
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")/.."
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOTDIR="$(git -C "$SCRIPTDIR" rev-parse --show-toplevel)"
+
+# ensure that the sort order is not locale-dependent
+export LC_ALL=C.UTF-8 
+
+set -x
 
 # see also ".mailmap" for how email addresses and names are deduplicated
+tee "${ROOTDIR}/AUTHORS" <<-EOF
```

Comment:
I just gave the updated script a try, but it looks like a downside of the new version is that it dumps all output when running it; previously, it would generate the script, but my terminal would not output 🤔

Reply:
Same as the `./generate-go-mod.sh` change in the other PR; this is intended to make debugging this easier/more obvious (especially in CI, which would be a follow-up). I'm happy to walk it back if you think it's too noisy.